fake_xml_http_request.js

```diff
@@ -447,6 +447,12 @@
         } catch (e) {
           // Unable to parse XML - no biggie
         }
+      } else if (this.responseText && /(application\/json)|(application\/vnd\.api\+json)/.test(type)) {
+        try {
+          this.response = JSON.parse(this.responseText);
```

Comment:
This isn't correct in terms of the spec. There is always a `response` no matter the `responseType`, but there is only a `responseText` when response type is `text`. `response` should be whatever the response type called for: string for `text`, xml for `xml`, object for `json`, etc.

https://xhr.spec.whatwg.org/#the-response-attribute

Reply:
@nlfurniss thanks, yes we should be careful to not deviate from the spec.

Could you briefly describe the implications of this violation? I suspect you have a use-case in-mind, and rather then guessing what the use-case is, knowing it will help the rest of us think about this correctly.

I likely have some follow up questions.

Also, do you see a path forward for the PR, or does your concern invalidate it entirely?

Reply:
My use-case is JSON, but it seemed like this semi-completeness might confuse people.

After looking at it again, I don't think it's a blocker but I'll create an issue after it lands setting out that `response` is not 100% supported